### ISSUE TITLE
"metacompleter" i.e. completing from a group of completers

### ISSUE THREAD

**Issue by sanga** [jonathanslenders/python-prompt-toolkit#630]
In prompt toolkit 1 I had a set up where I had a metacompleter that had a list of completer instances and then get_completions method of the metacompleter basically just yielded from all the it's completers. The reason for this is basically to support completion plugins that may or may not be loaded by the user.

This doesn't appear to work in prompt toolkit 2. Or at least I haven't gotten it to work in a short amount of time (it's possible that that is actually fine but that there's been some other API change that is causing to me to not see any completions).
```

class MetaCompleter(Completer):
    def __init__(self, *completer_instances):
       self.completers = completer_instances

     def get_completions(self, document, complete_event):
        for c in self.completers:
            for h in c.get_completions(document, complete_event):
                 yield h
```

Basically something like that ^. So is there a canonical way to do that with prompt toolkit? And should that work in prompt toolkit 2 or has there been changes in that area (I hadn't spotted anything obvious so far)

**Comment by jonathanslenders** [comment]
Hi @sanga,

I added a ``merge_completers`` function that can combine multiple completers into one.
See: https://github.com/jonathanslenders/python-prompt-toolkit/commit/d15d79ad6df9ea22f86b0179d04e08fbf602138d and https://github.com/jonathanslenders/python-prompt-toolkit/commit/4a431a999617683ae0cf4c7a2f2cd8e9d3a1d170 (the second commit contains an example).

Would this work for you?

**Comment by sanga** [comment]
Yes, that works. Thank you. I figured out what my problem was btw. Prompt tk has always said that all strings must be unicode however apparently prompt tk 2 got more strict about this (I was passing in regular strings in a couple of places and that used to work in v1). Anyway, that merge_completers func works just fine. Cheers!

**Comment by jonathanslenders** [comment]
Ok, good to hear!
